samples/boards/sensortile_box/app.overlay

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	chosen {
+		zephyr,console = &cdc_acm_uart1;
+	};
+
+	cdc_acm_uart1: cdc_acm_uart@1 {
+		compatible = "zephyr,cdc-acm-uart";
+		reg = <0x1 0x1>;
+		label = "CDC_ACM_1";
+	};
```

Comment:
@jfischer-no Let's bring the discussion we started offline about this into the PR.

Problem statement: this way of representing the CDC ACM configuration in devicetree does not create a link between the zephyr,cdc-acm-uart node and the node which represents the USB device controller hardware. This creates ambiguity if multiple USB device controllers are present and enabled in the same system. Specifically the ambiguity is: which USB device should be used for CDC ACM communication?

You told me offline that there is a plan to add support for multiple USB controllers to Zephyr's USB subsystem. Therefore, I think we should solve this problem to avoid forcing users to change their devicetrees later.

I can see two options:

1. Make these zephyr,cdc-acm-uart nodes children of the USB node. Example for Nordic:

```
&usbd {
	cdc_acm_uart1: cdc_acm_uart@1 {
		compatible = "zephyr,cdc-acm-uart";
		reg = <0x1 0x1>;
		label = "CDC_ACM_1";
	};
};
```

Then I think we should add `bus: usb` to the bindings for each USB controller. E.g. add `bus: usb` to nordic,nrf-usbd.yaml.

Then from the CDC ACM driver you can say:

```
#define DT_DRV_COMPAT zephyr_cdc_acm_uart

...
DT_BUS(DT_DRV_INST(inst))
```

if you need a node identifier for the USB controller.

2. With a phandle. E.g. something like this:

```
	cdc_acm_uart1: cdc_acm_uart@1 {
		compatible = "zephyr,cdc-acm-uart";
		reg = <0x1 0x1>;
		label = "CDC_ACM_1";
		usb-device = <&usbd>;
	};
```

And adding a `usb-device` property to the binding, with `type: phandle`.

And then use `DT_PHANDLE(DT_DRV_INST(inst), usb_device)` to get the node identifier for the USB controller.


What do you think? Any other ideas? @galak?

Reply:
Property name `usb-device` is unfortunate in USB nomenclature. I would opt for `controller`.

Reply:
I follow the approach "Make these zephyr,cdc-acm-uart nodes children of the USB node"